lib/schedulers/serialDispatcher.js

```diff
@@ -16,17 +16,23 @@ class SerialDispatcher {
             buffer: [],
             intervalRef: setIntervalAsync(async () => {
 
-                const ctx = this._queues.get(queue);
-                const delegate = context.buffer.shift();
+                try {
+                    const ctx = this._queues.get(queue);
+                    const delegate = context.buffer.shift();
 
-                if (delegate)  {
-                    await delegate();
+                    if (delegate)  {
+                        await delegate();
 
-                    if (!ctx.buffer.length) {
-                        this._queues.delete(queue);
-                        clearIntervalAsync(ctx.intervalRef);
+                        if (!ctx.buffer.length) {
+                            this._queues.delete(queue);
+                            clearIntervalAsync(ctx.intervalRef);
+                        }
                     }
                 }
+                catch (err) {
+                    console.error(err);
```

Comment:
@WesTyler Do you think it's alright that I used a console.error here in order to avoid a breaking change? Should I instead use the `$.logger` or let it throw the error?